setup.cfg

```diff
@@ -80,7 +80,6 @@ ignore-words-list = alers
 [mypy]
 exclude = ^\.git/|^__pycache__/|^docs/source/conf.py$|^old/|^build/|^dist/|\.tox
 python_version = 3.9
-install_types = True
```

Comment:
Perhaps instead add `non-interactive = True`
* https://mypy.readthedocs.io/en/stable/command_line.html#miscellaneous

Reply:
Potentially. Although I think we'd ideally not have mypy install any type hint packages. I'm still puzzled why it even tries. That might indicate that something else is wrong.

Reply:
You can either tell it everything (constant human effort to align type hints) or allow it to guess.

Reply:
Yeah, but we already have the type hint packages in `.[types]`, and they only change when our dependencies do. Here, it tries to install `types-requests`, which is already getting installed (via `.[all]`), yet somehow mypy can't find it...

Reply:
What happens if you modify line 69 to exactly match https://pypi.org/project/types-requests ?

Reply:
Not sure what you mean. The version dependency on line 69 of `setup.cfg` matches the one on line 30 for `requests` (so that it should install matching versions), and we can see that it does install the latest PyPI version [here](https://github.com/jjjake/internetarchive/actions/runs/7893419666/job/21542016167#step:5:168).

Quoting for future reference since the Actions logs expire stupidly quickly:

```
Run pip install .[all]
<...>
Collecting requests<3.0.0,>=2.25.0 (from internetarchive==3.6.0)
  Downloading requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
<...>
Collecting types-requests<3.0.0,>=2.25.0 (from internetarchive==3.6.0)
  Downloading types_requests-2.31.0.20240125-py3-none-any.whl.metadata (1.8 kB)
<...>
Downloading requests-2.31.0-py3-none-any.whl (62 kB)
<...>
Downloading types_requests-2.31.0.20240125-py3-none-any.whl (14 kB)
<...>
Installing collected packages: <...>, types-requests, <...>, requests, <...>
Successfully installed <...> requests-2.31.0 <...> types-requests-2.31.0.20240125 <...>
```